spectrum_io/search_result/search_results.py

```diff
@@ -1,7 +1,8 @@
 import logging
 import os
```

Comment:
get rid of this. This is used for the function generate_internal, where out_path is still only a string. Make this a path as well and use pathlib instead os. In fact, you can change line 44 to use pathlib's with_suffix method here